src/bodies/polygon.ts

```diff
@@ -71,6 +72,9 @@ export class Polygon extends SATPolygon implements BBox, Collider {
     | Types.Ellipse
     | Types.Line = Types.Polygon;
 
+  private pointsBackup!: SATVector[];
+  private scaleVector: SATVector;
```

Comment:
```suggestion
  private scaleVector: Vector = { x: 1, y: 1 };
```

simpler type will be faster propably